sec_certs/dataset.py

```diff
@@ -1037,44 +1054,55 @@ def compare_certs(current_certificate: 'FIPSCertificate', other_id: str):
         for current_cert in self.certs.values():
             if not current_cert.state.txt_state:
                 continue
-            for rule in current_cert.processed.keywords['rules_cert_id']:
-                for cert in current_cert.processed.keywords['rules_cert_id'][rule]:
-                    cert_id = ''.join(filter(str.isdigit, cert))
 
-                    if cert_id == '' or cert_id not in self.certs:
-                        broken_files.add(current_cert.dgst)
-                        current_cert.state.file_status = False
-                        break
-
-        if broken_files:
-            logger.warning("CERTIFICATE FILES WITH WRONG CERTIFICATES PARSED")
-            logger.warning(broken_files)
-            logger.warning("... skipping these...")
-            logger.warning(f"Total non-analyzable files:{len(broken_files)}")
+            for rule in current_cert.processed.keywords['rules_cert_id']:
+                matches = current_cert.processed.keywords['rules_cert_id'][rule]
+                current_cert.processed.keywords['rules_cert_id'][rule] = [cert_id for cert_id in matches if
+                                                                          validate_id(current_cert,
+                                                                                      cert_id.replace('Cert.', '')
+                                                                                      .replace('cert.', '')
+                                                                                      .lstrip("#CA0 "))
+                                                                          and cert_id != current_cert.cert_id]
 
         for current_cert in self.certs.values():
```

Comment:
Quite high cyclomatic complexity. For a starter, I suggest you implement a `@staticmethod` that operates on certificate and refactor the method, it is far too complex.